app/Http/Controllers/Three/ReviewsController.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Rogue\Http\Controllers\Three;
+
+use Rogue\Models\Post;
+use Illuminate\Http\Request;
+use Rogue\Repositories\Three\PostRepository;
+use Rogue\Http\Controllers\Api\ApiController;
+use Illuminate\Auth\Access\AuthorizationException;
+use Rogue\Http\Transformers\Three\PostTransformer;
+
+class ReviewsController extends ApiController
+{
+    /**
+     * The post repository instance.
+     *
+     * @var Rogue\Repositories\Three\PostRepository
+     */
+    protected $post;
+
+    /**
+     * @var \Rogue\Http\Transformers\Three\PostTransformer
+     */
+    protected $transformer;
+
+    /**
+     * Create a controller instance.
+     *
+     * @param  PostContract $posts
+     * @return void
+     */
+    public function __construct(PostRepository $post)
+    {
+        $this->post = $post;
+        $this->transformer = new PostTransformer;
+
+        $this->middleware('auth:api');
+    }
+
+    /**
+     * Update a post(s)'s status when reviewed.
+     *
+     * @param Request $request
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function reviews(Request $request)
+    {
+        $request->validate([
+            'post_id' => 'required',
+            'status' => 'in:pending,accepted,rejected',
+        ]);
+
+        // Only allow an admin to review the post.
+        if (token()->role() === 'admin') {
```

Comment:
We can handle this with the `role:admin` middleware, like we do elsewhere!

```diff
    /**
     * Create a controller instance.
     *
     * @param  PostContract $posts
     * @return void
     */
    public function __construct(PostRepository $post)
    {
        $this->post = $post;
        $this->transformer = new PostTransformer;

        $this->middleware('auth:api');
+       $this->middleware('role:admin');
    }
```

Reply:
oh right!